core/src/main/java/org/elasticsearch/action/admin/indices/rollover/Condition.java

```diff
@@ -49,6 +54,12 @@ protected Condition(String name) {
 
     public abstract Result evaluate(Stats stats);
 
+    /**
+     * Checks if this condition is available in a specific version.
+     * This makes sure BWC when introducing a new condition which is not recognized by older versions.
+     */
+    abstract boolean includedInVersion(Version version);
```

Comment:
maybe just default it to `true` then you don't need to override it in the existing ones

Reply:
@s1monw, I made it `abstract` to make sure that when we add new conditions, we have to pay attention for BWC. Should I update as you suggested or keep as it is? WDYT?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/27160/commits/c33d4b35bbaa7779ddcb754a3f83b51e3d8a1f15